### PR TITLE
[Accessibility] Add novalidate attribute to login form

### LINKED ISSUE #51824: Accessibility: Non-descriptive error message
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The error message "Please fill out this field" doesn't provide meaningful information about the failures, errors location or instructions on how to fix them.

T002_NonDescriptiveError

### Component or Page: 

Front-end Login Form on My Account Page

### Name affected element:

1. Username or email address, edit field

### Relevant code snippets:

`<input type="text" class="woocommerce-Input woocommerce-Input--text input-text" name="username" id="username" autocomplete="username" value="" required="" aria-required="true">`

### Recommended fix:

Provide a summary in the error message to indicate the input fields containing the errors and, if possible, instructions on how to fix it or the type of input that is required from the user. Your error message should allow the user to fix the error properly. For example, 'There was a problem with your submission. There are three problems to fix in the form: Name: You must provide a name, etc.

### LINKED ISSUE #51829: Accessibility: Critical information disappears after a time interval
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

Error message 'Please fill out this field': This element has critical functionality that disappears quickly before a user may be able to interact with it. Some users may need more time to think, react, or to navigate to the functionality and will be unable to interact with the element before it disappears.

T002_ErrorDisappears

### Component or Page: 

Front-end Login Form on My Account Page

### Name affected element:

Error message 'Please fill out this field'

### Relevant code snippets:

`<input type="text" class="woocommerce-Input woocommerce-Input--text input-text" name="username" id="username" autocomplete="username" value="" required="" aria-required="true">`

### Recommended fix:

Make sure critical , e.g., errors, warnings, or success messages, are always available and do not disappear after a few seconds. It is recommended that these messages stay present on the screen and instead of removing them after a time, you provide a dismiss button so users can choose to close them or they just stay visible until the form reauthenticates. If you want to have messages automatically disappear, you should provide a user preferences setting that allows users to adjust the time messages stay on the screen before disappearing.

### PR DESCRIPTION
**Pull request by darrenjacoby**
### Submission Review Guidelines:

- * I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- * I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- * I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
- * Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #51824
Closes #51829

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Browse to the login page.
2. Leave both the username/email address and password field blank and attempt to login.
3. Only server side validation should be shown (Chrome default client side validation is disabled via attribute `novalidate`).

### Changelog entry

- * [ ] Automatically create a changelog entry from the details below.
- * [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Type

</details>

### CODE REVIEW

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by qasumitbagthariya** [comment]
### QA/Test Report ✅ 
-------------

I have checked this PR in the `darrenjacoby:fix/51829` branch and changes working as expected.

**Additional Check:**

Test this behavior on different browsers (Chrome, Firefox, Safari, etc.) to ensure the novalidate attribute is working consistently across all of them.

**Testing Environment** 

<Details>

- WordPress: 6.7
- Theme: Storefront 4.6.0
- WooCommerce - WooCommerce - Dev version with this fix specific branch.
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Git Branch: darrenjacoby:fix/51829

</Details>


https://github.com/user-attachments/assets/1d337f6f-6769-49e9-8387-4817b5af458b



**chrome**

image

**firefox**
image

@darrenjacoby 
please remove [WIP] from the title and take this PR out of draft.

status update:
@vikrampm1

**Comment by vikrampm1** [comment]
@darrenjacoby please remove [WIP] from the title and take this PR out of draft. Otherwise, @nielslange this PR is ready for your review!

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @darrenjacoby. I've successfully tested the PR and will merge it once all CI checks passed.

**Comment by vikrampm1** [comment]
@nielslange just checking if there is any additional help needed here or if this good to be merged.

**Comment by nielslange** [comment]
> @nielslange just checking if there is any additional help needed here or if this good to be merged.

Thanks for the ping, @vikrampm1. Apparently, this issue slipped through. I'll take care of it now.

File: plugins/woocommerce/templates/myaccount/form-login.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
  *
  * @see     https://woocommerce.com/document/template-structure/
  * @package WooCommerce\Templates
- * @version 9.2.0
+ * @version 9.6.0
```
Comment (nielslange) [inline plugins/woocommerce/templates/myaccount/form-login.php]:
```suggestion
 * @version 9.7.0
```

I'm bumping this to `9.7.0`, as the code freeze for `9.6.0` was a few days ago.

**Comment by nielslange** [comment]
Due to a recent change in the nightly build of WordPress, [one of the PHPUnit tests](https://github.com/woocommerce/woocommerce/actions/runs/12406126939/job/34634910009?pr=53386) keeps failing. This issue should be resolved with #53864.

**Pull request merged**